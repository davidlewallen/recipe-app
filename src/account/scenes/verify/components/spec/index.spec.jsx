import React from 'react';
import { shallow } from 'enzyme';
import VerifyEmail, { Verifying, Verified, Resend, Resent, Invalid } from '..';

describe('VerifyEmail Component', () => {
  let wrapper = null;
  let mockProps = {};

  beforeEach(() => {
    mockProps = {
      verificationState: '',
      handleResendVerification: jest.fn(),
    };

    wrapper = shallow(<VerifyEmail {...mockProps} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Verifying /> if verificationState = "checking"', () => {
    wrapper.setProps({ verificationState: 'checking' });

    expect(wrapper.find('Verifying').exists()).toBe(true);
  });

  it('should render <Verified /> if verificationState = "verified"', () => {
    wrapper.setProps({ verificationState: 'verified' });

    expect(wrapper.find('Verified').exists()).toBe(true);
  });

  it('should render <Resend /> if verificationState = "resend"', () => {
    wrapper.setProps({ verificationState: 'resend' });

    expect(wrapper.find('Resend').exists()).toBe(true);
  });

  it('should render <Invalid /> if verificationState = "nonMatching"', () => {
    wrapper.setProps({ verificationState: 'nonMatching' });

    expect(wrapper.find('Invalid').exists()).toBe(true);
  });
  it('should render <Resent /> if verificationState = "resent"', () => {
    wrapper.setProps({ verificationState: 'resent' });

    expect(wrapper.find('Resent').exists()).toBe(true);
  });
});

describe('Verifying', () => {
  it('should matchsnapshot', () => {
    expect(shallow(<Verifying />)).toMatchSnapshot();
  });
});

describe('Verified', () => {
  it('should matchsnapshot', () => {
    expect(shallow(<Verified />)).toMatchSnapshot();
  });
});

describe('Resend', () => {
  it('should matchsnapshot', () => {
    expect(
      shallow(<Resend handleResendVerification={jest.fn()} />)
    ).toMatchSnapshot();
  });
});

describe('Resent', () => {
  it('should matchsnapshot', () => {
    expect(shallow(<Resent />)).toMatchSnapshot();
  });
});

describe('Invalid', () => {
  it('should matchsnapshot', () => {
    expect(shallow(<Invalid />)).toMatchSnapshot();
  });
});
