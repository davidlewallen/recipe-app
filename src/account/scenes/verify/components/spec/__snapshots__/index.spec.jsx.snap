// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid should matchsnapshot 1`] = `
<Fragment>
  <p>
    Your verification key did not match.
  </p>
  <p>
    Please contact support.
  </p>
</Fragment>
`;

exports[`Resend should matchsnapshot 1`] = `
<Fragment>
  <p>
    Your email verification expired. Click below to resend verification email.
  </p>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Resend Verification Email
  </Button>
</Fragment>
`;

exports[`Resent should matchsnapshot 1`] = `
<p>
  To access your account, please follow the steps included in the email that was just sent to you.
</p>
`;

exports[`Verified should matchsnapshot 1`] = `
<Fragment>
  <p>
    Thank you for verifying your email.
  </p>
  <p>
    You will be redirected to the login page in 5 seconds.
  </p>
</Fragment>
`;

exports[`VerifyEmail Component should render 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
>
  <Uncontrolled(Panel)>
    <PanelBody
      bsClass="panel"
      className="align-center"
      collapsible={false}
    />
  </Uncontrolled(Panel)>
</Grid>
`;

exports[`Verifying should matchsnapshot 1`] = `
<p>
  Verifying email address...
</p>
`;
